viadot/sources/cloud_for_customers.py

```diff
@@ -0,0 +1,256 @@
+        self.query_endpoint = endpoint
+
+        if filter_params:
+            filter_params_merged = self.DEFAULT_PARAMS.copy()
+            filter_params_merged.update(filter_params)
+
+            self.filter_params = filter_params_merged
+        else:
+            self.filter_params = self.DEFAULT_PARAMS
+
+        if self.url:
+            self.full_url = urljoin(self.url, self.query_endpoint)
+
+    @staticmethod
+    def create_metadata_url(url: str) -> str:
+        """Creates url to fetch metadata to.
+
+        Args:
+            url (str): The url to transform to metadata url.
+
+        Returns:
+            meta_url (str): The url to fetch metadata with.
+        """
+        start = url.split(".svc")[0]
+        url_raw = url.split("?")[0]
+        end = url_raw.split("/")[-1]
+        meta_url = start + ".svc/$metadata?entityset=" + end
+        return meta_url
+
+    def _extract_records_from_report_url(self, report_url: str) -> List[Dict[str, Any]]:
+        """Fetches report_url to exctract records.
+
+        Args:
+            report_url (str): The url to extract records from.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from url.
+        """
+        records = []
+        while report_url:
+            response = self.get_response(report_url, filter_params=self.filter_params)
+            response_json = response.json()
+            new_records = self.get_entities(response_json, report_url)
+            records.extend(new_records)
+
+            report_url = response_json["d"].get("__next")
+
+        return records
+
+    def _extract_records_from_url(self, url: str) -> List[Dict[str, Any]]:
+        """Fetches url to exctract records.
+
+        Args:
+            url (str): The url to extract records from.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from url.
+        """
+        tmp_full_url = deepcopy(url)
+        tmp_filter_params = deepcopy(self.filter_params)
+        records = []
+        while url:
+            response = self.get_response(tmp_full_url, filter_params=tmp_filter_params)
+            response_json = response.json()
+            if isinstance(response_json["d"], dict):
+                # ODATA v2+ API
+                new_records = response_json["d"].get("results")
+                url = response_json["d"].get("__next", None)
+            else:
+                # ODATA v1
+                new_records = response_json["d"]
+                url = response_json.get("__next", None)
+
+            # prevents concatenation of previous url's with filter_params with the same filter_params
+            tmp_filter_params = None
+            tmp_full_url = url
+
+            if hasattr(new_records, "__iter__"):
+                records.extend(new_records)
+        return records
+
+    def extract_records(self) -> List[Dict[str, Any]]:
+        """Downloads records from url or report_url if present.
```

Comment:
If we're referencing variables I think it's more clear to use the backticks

Reply:
Done